### DISCUSSION TITLE
Actividad 4 - Grupo 12

### DISCUSSION THREAD

**Discussion by NicoPavoni** [frlp-utn-ingsoft/catedra#160] [Show and tell]
https://github.com/NicoPavoni/vetsoft

**Comment by RodrigoJacznik** [comment]
hola. Estaba revisando esta actividad. Como hicieron para correr los test de unidad y de integración? No estoy pudiendo

**Reply by ianbifano** [reply]
Hola Rodri, los tests de unidad y de integracion estan en la carpeta app/tests. No se si todos los grupos hicieron los tests asi, quizas estas apuntando a otro directorio.

**Reply by RodrigoJacznik** [reply]
Mi pregunta va mas que nada porque me baje el proyecto y no me corren. Tambien revise su ci y no corrio ningún test de unidad e integracion.
Que comando usaron para correrlos?

**Reply by NicoPavoni** [reply]
Hola @RodrigoJacznik , los test de integración y unidad funcionan. Te dejo 2 screens para que puedas ver como correrlos.

Ejemplo test de integracion:

![image](https://github.com/frlp-utn-ingsoft/catedra/assets/104012558/0f54a933-d6f8-4766-91f8-a82bfa8ed063)

Ejemplo test de unidad : 

![image](https://github.com/frlp-utn-ingsoft/catedra/assets/104012558/917823f4-76b7-46ac-9783-1384257697eb)

**Reply by RodrigoJacznik** [reply]
Claro. Pero en ese caso tengo que ir uno por uno.
Fíjense que en el ci no les corre ningún test. Algo que está mal ya que aceptaron prs sin esa comprobación.

**Reply by NicoPavoni** [reply]
Hola @RodrigoJacznik perdón la demora, ya está resuelto.

**Comment by RodrigoJacznik** [comment]
Grupo 12

Buenas. Les dejo el feedback de la actividad 3 y 4.

Actividad 3:

Los felicito por la descripción de los PRs. Creo que los títulos podrían estar mejor. Me gusto la interacción que tuvieron en algunos PRs
La decisión que tomaron al separar la feature de los test en branch distintos no me pareció correcta. En general la feature va acompañada con sus test correspondintes.

Lo ideal seria este PR: https://github.com/NicoPavoni/vetsoft/pull/12

Recuerden que un PR tiene que ser atómico. Pe: https://github.com/NicoPavoni/vetsoft/pull/1 tiene un test e2e que no tiene nada que ver con la feature.

@ianbifano, vi que hiciste muchos test, pero no encontré cual fue la tarea que tomaste. Me pasas el link al PR?

Actividad 4:

Buenas. Encontramos algunos problemas:

- La tarea 'Prepare coverage' no esta corriendo ningun test, por eso el coverage es tan bajo.
Tienen que cambiar el comando que se ejecuta en la tarea 'Run unit and integration tests' por 'coverage run --source="./app" --omit="./app/migrations/**" manage.py test app' y borrar la tarea prepare coverage

- Tengan cuidado con los PRs no es atomico como este https://github.com/NicoPavoni/vetsoft/pull/17 Cambia las reglas de RUFF y cambia CI

Rodriguez Luciano, no pude encontrar cual fue la actividad que realizaste. Me podes pasar el PR?
Bifano Ian. No realizaste ningun test sobre la tarea que realizaste
García Montes Luciano. No realizaste ningun test sobre la tarea que realizaste

Les voy a pedir que Arreglen el CI y luego:
Rodriguez Luciano, hagas la actividad faltante con por lo menos un test e2e, uno de integracion y uno unidad
Bifano Ian. Agregues por lo menos un test e2e, uno de integracion y uno unidad
García Montes Luciano. Agregues por lo menos un test e2e, uno de integracion y uno unidad

Tienen tiempo hasta el 14/7

**Reply by lucianotgm12** [reply]
Buenas tardes Rodrigo!

Respecto a la Actividad 3: @ianbifano hizo la validacion de la fecha de cumpleaños de la mascota que tenia realizada, y solo agregó los tests.

Respecto a la actividad 4: no los habíamos hecho porque pensamos que con corregir los tests que ya existían alcanzaba. No obstante ya están agregados. 

@LucianooRodriguez hizo la verificación de que los números de teléfono comiencen en 54. El PR que hizo inicialmente con esa actividad es [este.](https://github.com/NicoPavoni/vetsoft/pull/21)

Cualquier cosa por favor avisanos. Saludos!
